### PR TITLE
Add SSL support

### PR DESCRIPTION
**Pull request by kepten**
based on https://github.com/facebookarchive/scribe/pull/78

### CODE REVIEW

**Comment by kepten** [comment]
Aaand it looks like it's working:

``` sh
root@5505837ccb3a:/build/scribe# ps aux | grep scribe
root       156  0.0  0.3 187120  6876 ?        Sl   20:09   0:00 scribed -c examples/exampleSSLcentral.conf
root      1631  0.4  0.3 252556  6756 ?        Sl   20:36   0:00 scribed -c examples/exampleSSLclient.conf
root@5505837ccb3a:/build/scribe# echo 'TEST' | ./examples/scribe_apache  kepten_category
[Wed Aug 10 20:36:57 2016] "[kepten_category] Creating new category store from model default"
[Wed Aug 10 20:36:57 2016] "store thread starting"
root@5505837ccb3a:/build/scribe# [Wed Aug 10 20:36:57 2016] "SSL: Using <examples/cert/client.crt> for cert and <examples/cert/client.key> for the key"
[Wed Aug 10 20:36:57 2016] "SSL: Using <examples/cert/server.crt> as the trusted list of certs"
[Wed Aug 10 20:36:57 2016] "SSL: Requiring remote side have a valid cert too"
[Wed Aug 10 20:36:57 2016] "Opened connection to remote scribe server <localhost:1464>"
[Wed Aug 10 20:36:58 2016] "Opened connection to remote scribe server <localhost:1464>"
Thrift: Wed Aug 10 20:36:58 2016 TConnectedClient died: SSL_accept: error code: 0
[Wed Aug 10 20:36:58 2016] "[kepten_category] Creating new category store from model default"
[Wed Aug 10 20:36:58 2016] "store thread starting"
[Wed Aug 10 20:36:58 2016] "Successfully sent <1> messages to remote scribe server <localhost:1464>"
[Wed Aug 10 20:36:58 2016] "[kepten_category] Opened file </tmp/scribetest/kepten_category/kepten_category_00000> for writing"
```

However I had to manually install `python-six` and symlink compiled python lib directories to `/usr/lib/python2.7/dist-packages/` as follow. These should be handled by the .deb packages I guess.

```
root@5505837ccb3a:/build/scribe# ll /usr/lib/python2.7/dist-packages/
total 116
drwxr-xr-x  4 root root  4096 Aug 10 20:33 ./
drwxr-xr-x 30 root root  4096 Aug 10 20:24 ../
drwxr-xr-x  2 root root  4096 Aug 10 18:38 Pyste/
lrwxrwxrwx  1 root root    51 Jan 16  2014 Pyste-0.9.10.egg-info -> ../../../share/pyshared/Pyste-0.9.10-py2.7.egg-info
-rw-r--r--  1 root root   119 Jun 22  2015 README
drwxr-xr-x  2 root root  4096 Aug 10 18:38 boost/
-rw-r--r--  1 root root  5971 Feb 23  2014 debconf.py
-rw-r--r--  1 root root  6013 Aug 10 18:38 debconf.pyc
lrwxrwxrwx  1 root root    22 Aug 10 20:31 fb303 -> ../site-packages/fb303/
-rw-r--r--  1 root root 13648 May 19  2013 lsb_release.py
-rw-r--r--  1 root root 10284 Aug 10 18:38 lsb_release.pyc
lrwxrwxrwx  1 root root    23 Aug 10 20:24 scribe -> ../site-packages/scribe/
-rw-r--r--  1 root root  1418 Mar 26  2015 six-1.5.2.egg-info
-rw-r--r--  1 root root 22857 Jan  6  2014 six.py
-rw-r--r--  1 root root 22317 Aug 10 20:33 six.pyc
lrwxrwxrwx  1 root root    23 Aug 10 20:30 thrift -> ../site-packages/thrift/
root@5505837ccb3a:/build/scribe# ll /usr/lib/python2.7/site-packages/
total 28
drwxr-xr-x  5 root root 4096 Aug 10 20:29 ./
drwxr-xr-x 30 root root 4096 Aug 10 20:24 ../
drwxr-xr-x  2 root root 4096 Aug 10 20:11 fb303/
drwxr-xr-x  2 root root 4096 Aug 10 20:11 fb303_scripts/
drwxr-xr-x  2 root root 4096 Aug 10 20:02 scribe/
-rw-r--r--  1 root root  178 Aug 10 19:56 scribe-2.0-py2.7.egg-info
lrwxrwxrwx  1 root root   24 Aug 10 20:29 thrift -> /build/thrift/lib/py/src/
-rw-r--r--  1 root root  609 Aug 10 19:56 thrift_fb303-1.0.0_dev-py2.7.egg-info
```

**Comment by zsol** [comment]
Awesome

File: src/Makefile.am [hunk 1]
```diff
@@ -34,7 +34,7 @@ all:
 #
 THRIFT = $(thrift_home)/bin/thrift
 THRIFT_INCLUDES = -I $(fb303_home)/share/
-THRIFT_OPTS = $(THRIFT_INCLUDES) --gen cpp:pure_enums --gen py --gen php --gen java
+THRIFT_OPTS = $(THRIFT_INCLUDES) --gen cpp:pure_enums --gen py --gen php --gen java --gen erl
```
Comment (abesto) [inline src/Makefile.am]: Why do we need erlang?

**Comment by abesto** [comment]
Why did you need to install `python-six`?

**Comment by abesto** [comment]
BTW: if there's a .deb package for `python-six`, then we can just add a dependency to it via `fpm`. If not, then we'll probably need a (post?) install script, I'm pretty sure that's doable with `fpm` as well.

File: src/Makefile.am (hunk 1 above)
Comment (kepten) [inline src/Makefile.am]: I personally don't but it was in the branch. :P

**Comment by kepten** [comment]
There is an APT package for `python-six`, so it's easy to add it as dependency yes (btw `scribe_apache` needs it) but I think the Python bindings (thrift, scribe, etc.) should be included in their respective .deb packages.

**Comment by abesto** [comment]
Agree that the Python binding should be included. For the purposes of this PR, I think we can merge once you add the dependency on `python-six`.

**Comment by abesto** [comment]
Published internally as `thrift_1.0.0-dev~20160815~prezi4_amd64`, `thrift-fb303_1.0.0-dev~20160815~prezi4_amd64.deb`, and `scribe_0.2.2~20160815~prezi4_amd64`

**Pull request merged**